### PR TITLE
Implement initial REST API

### PR DESCRIPTION
**Pull request by ajoaoff**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -40,3 +47,107 @@ def shutdown(self):
...
+            elif now > mw.end:
+                result = {'response': f'Maintenance window {mw_id} has '
+                                      f'already finished.'}
+                status = 400
+            else:
+                self.scheduler.remove(mw)
+                mw.end_mw()
+                result = {'response': f'Maintenance window {mw_id} finished.'}
+                status = 200
+        return jsonify(result), status
```
Comment (hdiogenes) [inline main.py:153]: For all the REST methods I'd suggest copying the `return jsonify(...)` to the top of the function (inside the error code path) and moving all the "default" code path to outside the `else` block.

> flat is better than nested

File: models.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                return None
+            items.append(item)
+        return cls(start, end, controller, items, mw_id)
+
+    def update(self, mw_dict, controller):
+        """Update a maintenance window with the data from a dictionary."""
+        if 'start' in mw_dict:
+            self.start = datetime.datetime.strptime(mw_dict['start'], TIME_FMT).astimezone(pytz.utc)
+        if 'end' in mw_dict:
+            self.end = datetime.datetime.strptime(mw_dict['end'], TIME_FMT).astimezone(pytz.utc)
```
Comment (hdiogenes) [inline models.py:79]: Suggestion: create small helper functions for this conversion.

File: models.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            items = list()
+            for i in mw_dict['items']:
+                try:
+                    item = self.uni_from_dict(i, controller)
+                except KeyError:
+                    item = self.link_from_dict(i, controller)
+                except TypeError:
+                    item = i
+                if item:
+                    items.append(item)
```
Comment (hdiogenes) [inline models.py:90]: This code is also duplicated, and should also be turned into a helper function.

File: models.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            link.extend_metadata(link_dict['metadata'])
+        s_vlan = link.get_metadata('s_vlan')
+        if s_vlan:
+            tag = TAG.from_dict(s_vlan)
+            link.update_metadata('s_vlan', tag)
+        return link
+
+    def start_mw(self):
+        """Actions taken when a maintenance window starts."""
+        pass
```
Comment (hdiogenes) [inline models.py:128]: I guess the linter will complain about 'unnecessary-pass'.

Comment (ajoaoff) [reply models.py:128]: This is something that will be implemented later, but I needed the function defined.

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+# Kytos var folder
+VAR_PATH = BASE_ENV / 'var' / 'lib' / 'kytos'
+# Path for enabled NApps
+ENABLED_PATH = VAR_PATH / 'napps'
+# Path to install NApps
+INSTALLED_PATH = VAR_PATH / 'napps' / '.installed'
+CURRENT_DIR = Path('.').resolve()
+
+# NApps enabled by default
+CORE_NAPPS = ['of_core']
```
Comment (hdiogenes) [inline setup.py:39]: Is `of_core` really required?

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+from setuptools.command.install import install
+
+if 'bdist_wheel' in sys.argv:
+    raise RuntimeError("This setup.py does not support wheels")
+
+# Paths setup with virtualenv detection
+if 'VIRTUAL_ENV' in os.environ:
+    BASE_ENV = Path(os.environ['VIRTUAL_ENV'])
+else:
+    BASE_ENV = Path('/')
```
Comment (hdiogenes) [inline setup.py:25]:
```suggestion
BASE_ENV = Path(os.environ.get('VIRTUAL_ENV', '/'))
```

**Pull request merged**